### PR TITLE
Add thread ID to step commands

### PR DESCRIPTION
**Pull request by thegecko**
Step commands optionally support threadIDs. This PR adds this support.

### CODE REVIEW

**Comment by thegecko** [comment]
Can this be merged?

File: src/mi/exec.ts [hunk 1]
```diff
@@ -20,18 +20,34 @@ export function sendExecRun(gdb: GDBBackend) {
...
+    }
+    return gdb.sendCommand(command);
 }
 
-export function sendExecNext(gdb: GDBBackend) {
-    return gdb.sendCommand('-exec-next');
+export function sendExecNext(gdb: GDBBackend, threadId?: number) {
+    let command = '-exec-next';
+    if (threadId) {
+        command += ` --thread ${threadId}`;
```
Comment (marechal-p) [inline src/mi/exec.ts]: According to [GDB MI docs](https://sourceware.org/gdb/download/onlinedocs/gdb/GDB_002fMI-Program-Execution.html), `-exec-next` doesn't support a `--thread` argument, unless I am mistaken?

Did you try it/is it actually working?

Comment (thegecko) [reply src/mi/exec.ts]: Yes, it's implemented in the adapter we use:

https://github.com/Marus/cortex-debug/blob/12bc8c0c4f17b11b1fb22e218b45732c521863bb/src/backend/mi2/mi2.ts#L274

Comment (dschaefer) [reply src/mi/exec.ts]: I guess the real question is does this work on all gdb's? Again, we need tests for this so we can prove that one way or another.

Comment (simark) [reply src/mi/exec.ts]: > According to [GDB MI docs](https://sourceware.org/gdb/download/onlinedocs/gdb/GDB_002fMI-Program-Execution.html), `-exec-next` doesn't support a `--thread` argument, unless I am mistaken?

All commands support it, it's documented once for all commands here:

https://sourceware.org/gdb/download/onlinedocs/gdb/Context-management.html#Context-management

Comment (marechal-p) [reply src/mi/exec.ts]: Thank you for the link!

**Comment by thegecko** [comment]
bump

**Comment by dschaefer** [comment]
Sorry, I've been on vacation this week. Will get to them all on Monday.

**Pull request merged**